### DISCUSSION TITLE
Validate the Node-locked license in different devices

### DISCUSSION THREAD

**Discussion by NoorAhmd** [Labs64/NetLicensing-Community#91] [Questions] [answered]
**Question**
How would you validate machine ID or device ID while validating the node-locked license.

**Additional context**
I am using the demo account to create a demo node-locked license. I followed every step to create the license and then use `POST https://go.netlicensing.io/core/v2/rest/licensee/IVJ24FYEJ/validate` with Postman. My License seems valid and I get the following response.

`<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<ns2:netlicensing xmlns="http://www.w3.org/2000/09/xmldsig#" xmlns:ns2="http://netlicensing.labs64.com/schema/context" ttl="2021-05-05T09:36:49.357Z">
    <ns2:infos/>
    <ns2:items>
        <ns2:item type="ProductModuleValidation">
            <ns2:property name="productModuleNumber">MPUGFD3WW</ns2:property>
            <ns2:property name="valid">true</ns2:property>
            <ns2:property name="productModuleName">1-nl</ns2:property>
            <ns2:property name="licensingModel">NodeLocked</ns2:property>
        </ns2:item>
    </ns2:items>
</ns2:netlicensing>`

My question is how I understand if the license only get validated or works in my Laptop only?

**Comment by r-brown** [comment, marked as answer]
Once customer got a [Node-Locked license](https://netlicensing.io/wiki/node-locked), license details view will show number of the available and paired session.

#### New customer / license

<img width="1070" alt="Screenshot 2021-05-05 at 21 09 09" src="https://user-images.githubusercontent.com/1361258/117196012-46646500-ade6-11eb-9cbd-b8f51e56f7da.png">

First validation request will pair customer number `IVJ24FYEJ` and device id `secret1` in the below example:

#### Request 1 - Initial validation and node pairing
```shell
curl -X POST -H "Authorization: Basic YXBpS2V5OjJjYmViYmJiLWIxNDAtNDRkOS05NzY3LTAzYTY1ZmE4YzhjZQ==" -H "Accept: application/xml" -H "Content-type: application/x-www-form-urlencoded" -d "productModuleNumber=MPUGFD3WW&nodeSecret=secret1" https://go.netlicensing.io/core/v2/rest/licensee/IVJ24FYEJ/validate
```

```xml
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<ns2:netlicensing xmlns="http://www.w3.org/2000/09/xmldsig#" xmlns:ns2="http://netlicensing.labs64.com/schema/context" ttl="2021-05-05T20:05:50.165Z"><ns2:infos/>
  <ns2:items>
    <ns2:item type="ProductModuleValidation">
      <ns2:property name="productModuleNumber">MPUGFD3WW</ns2:property>
      <ns2:property name="valid">true</ns2:property>
      <ns2:property name="productModuleName">1-nl</ns2:property>
      <ns2:property name="licensingModel">NodeLocked</ns2:property>
    </ns2:item>
  </ns2:items>
</ns2:netlicensing>
```

Customer's devices will be shown in the license datail view.

<img width="1078" alt="Screenshot 2021-05-05 at 21 09 19" src="https://user-images.githubusercontent.com/1361258/117195961-364c8580-ade6-11eb-8639-a9da4824fe93.png">

Attempt to use another device (for instance device with id `secret2`) will fail as the customer doesn't have enough nodes available.

#### Request 2 - Validation with the wrong nodeSecret
```shell
curl -X POST -H "Authorization: Basic YXBpS2V5OjJjYmViYmJiLWIxNDAtNDRkOS05NzY3LTAzYTY1ZmE4YzhjZQ==" -H "Accept: application/xml" -H "Content-type: application/x-www-form-urlencoded" -d "productModuleNumber=MPUGFD3WW&nodeSecret=secret2" https://go.netlicensing.io/core/v2/rest/licensee/IVJ24FYEJ/validate
```

```xml
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<ns2:netlicensing xmlns="http://www.w3.org/2000/09/xmldsig#" xmlns:ns2="http://netlicensing.labs64.com/schema/context" ttl="2021-05-05T20:06:09.138Z">
  <ns2:infos>
    <ns2:info id="NoFreeNodes" type="info">No free nodes found.</ns2:info>
  </ns2:infos>
  <ns2:items>
    <ns2:item type="ProductModuleValidation">
      <ns2:property name="productModuleNumber">MPUGFD3WW</ns2:property>
      <ns2:property name="valid">false</ns2:property>
      <ns2:property name="productModuleName">1-nl</ns2:property>
      <ns2:property name="licensingModel">NodeLocked</ns2:property>
    </ns2:item>
  </ns2:items>
</ns2:netlicensing>
```
